### PR TITLE
Added `ht.broadcast_shapes`

### PR DESCRIPTION
**Pull request by samadpls**
## Due Diligence

- General:
    - [X]  **base branch** must be `main` for new features, latest release branch (e.g. `release/1.3.x`) for bug fixes
    - [X]  **title** of the PR is suitable to appear in the [Release Notes](https://github.com/helmholtz-analytics/heat/releases/latest)
- Implementation:
    - [X] unit tests: all split configurations tested
    - [X] unit tests: multiple dtypes tested
    - [X] documentation updated where needed

## Description
This PR addresses issue #904  by implementing `ht.broadcast_shapes`to align with numpy and torch counterparts in terms of naming, calling, and handling multiple shapes.

Issue/s resolved: #904 

## Changes proposed:
- Added ht.broadcast_shapes with the desired behavior.

## Type of change

New feature (non-breaking change which adds functionality)
## Memory requirements
Memory requirements for this change have not been profiled at this time.

## Performance
Performance metrics for this change have not been measured at this time.

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

**Comment by samadpls** [comment]
Hey @ClaudiaComito, could you please review this whenever it's convenient for you?

File: heat/core/stride_tricks.py [hunk 1]
```diff
@@ -67,6 +67,61 @@ def broadcast_shape(shape_a: Tuple[int, ...], shape_b: Tuple[int, ...]) -> Tuple
...
+    except AttributeError:  # torch < 1.8
+        it = itertools.zip_longest(*shapes[::-1], fillvalue=1)
+        resulting_shape = max(len(shape) for shape in shapes) * [None]
+        for i, values in enumerate(it):
+            # if any value is 0, or if there is a mix of 1s and non-1s
+            if any(val == 0 or (val != 1 and val != values[0]) for val in values):
+                raise ValueError(f"operands could not be broadcast, input shapes {shapes}")
+            else:
+                resulting_shape[i] = max(values)
+        return tuple(resulting_shape[::-1])
```
Comment (ClaudiaComito) [inline heat/core/stride_tricks.py:116]: actually we stopped supporting torch<1.8 so this shouldn't be necessary

**Review by ClaudiaComito — CHANGES_REQUESTED** [review]
Wonderful @samadpls, just a small change needed.

**Review by ClaudiaComito — APPROVED** [review]
👏🏼 @samadpls thanks!

**Review by ClaudiaComito — APPROVED** [review]

**Pull request merged**